### DISCUSSION TITLE
`onValueChange` prop of a controlled `NumberInput` does not set value correctly in testing DOM

### DISCUSSION THREAD

**Discussion by spatial-andrewc** [chakra-ui/ark#2164] [General]
### Description

Since upgrading to `@ark-ui/react` major version 1 I am noticing that the `onValueChange` callback incorrectly sets the value of the input element in a jest DOM when using the `@tesing-library/react` and `@testing-library/user-event` libs. I have included a devbox that shows how the same test fails when using `onValueChange` but passes when using `onChange`.

The expected behaviour is that `onValueChange` correctly sets the input value.

### Link to Reproduction (or Detailed Explanation)

https://codesandbox.io/p/devbox/jolly-wood-gz5djk

### Steps to Reproduce

1. Go to devbox
2. Start the task called `test`
3. Run all test
4. See that one test passes and the other fails

### Ark UI Version

1.2.1

### Framework

- [X] React
- [ ] Solid
- [ ] Vue

**Comment by spatial-andrewc** [comment]
I've included another regression that occurs in the testing DOM in the provided code sandbox. This was noticed after the same upgrade.

```javascript
  it("will fail because it accepts text", async () => {
    render(<NumberFieldFailsWithText />);
    const numberInput = screen.getByRole("spinbutton");
    await user.type(numberInput, "~as*jhsqwr*");
    await waitFor(() => expect(numberInput).toHaveValue(""));
  });
```

```
Expected the element to have value:

    Received:
      ~as*jhsqwr*
```

**Comment by spatial-andrewc** [comment]
And another here

```javascript
it("will fail because it does not respect constraints", async () => {
    render(<NumberFieldPasses max={1} formatOptions={{ style: "percent" }}/>);
    const numberInput = screen.getByRole("spinbutton");
    await user.type(numberInput, "1000");
    fireEvent.blur(numberInput);
    await waitFor(() => expect(numberInput).toHaveValue("100%"));
  });
```

```
    expect(element).toHaveValue(100%)

    Expected the element to have value:
      100%
    Received:
      1000
```

**Comment by segunadebayo** [comment]
Closing — `onValueChange` properly returns both `value` and `valueAsNumber`. This has been fixed.
